docs/plugin-development.rst

```diff
@@ -508,6 +508,41 @@ login via a username and password. This allows alternative authentication
 modes to be used instead of core, or prior to attempting core authentication.
 The event info contains two keys, "login" and "password".
 
+Customizing the Swagger page
+****************************
+
+To customize text on the Swagger page, create a
+`Mako template <http://www.makotemplates.org/>`_ file that inherits from the
+base template and overrides one or more blocks:
+
+.. code-block:: html+mako
+
+    <%inherit file="${context.get('baseTemplateFilename')}"/>
+
+    <%block name="docsHeader">
+      <span>Cat programming interface</span>
+    </%block>
+
+    <%block name="docsBody">
+      <p>Manage your cats using the resources below.</p>
+    </%block>
+
+Install the template by registering its directory and setting its filename on
+``apiRoot``:
+
+.. code-block:: python
+
+    def load(info):
+        # Set base template filename variable
```

Comment:
This example is a little unclear. More comments would help to reflect the fact that:
1. When `info['apiRoot']` was already instantiated by Girder to render `api_docs.mako`, its template directories include the path to `api_docs.mako`
2. `info['apiRoot'].templateFilename` now just stores the string `'api_docs.mako'`
3. setting `info['apiRoot'].addTemplateDirectory(os.path.join(info['pluginRootDir'], 'server'))` is only necessary because the new `info['apiRoot'].templateFilename` is not fully qualified
4. due to 1., the effective lookup `<%inherit file="api_docs.mako"/>` will succeed

Perhaps this example could be restructured to better separate the two distinct steps happening here: the setting of a new `info['apiRoot'].templateFilename` and the use of `%inherit` to reference the old `api_docs.mako`.